tests/e2e/test_client.py

```diff
@@ -148,7 +148,7 @@ def test_internal_error(self):
         query.query_id = 9999999999999
 
         with self.assertRaises(DuneError) as err:
-            dune.execute(query)
+            dune.execute_query(query)
```

Comment:
Now, looking at the code and having refreshed my memory:

question for you:
- if we have a `QueryBase` or `Query` object, shouldn't we have the methods on that class:
- query.execute()
- query.result()
- etc.. ?

with functions like `dune.execute_query()` it feels more idiomatic this way:
```
res = dune.execute_query(query_id=1234)
```
instead of:
```
res = dune.execute_query(Query(1234))
```

for me, the two idiomatic options would be:
 * dune.execute_query(1234)  # or query_id=1234
 * dune.Query(1234).execute()

Right now, I see bits of OO in here, but inconsistently:
- Executions are simply `job_id`
- Queries are objects, with a query_id inside (it should be Query.id, right?)
  - but we don't have methods on Queries..
  - if we did, it would add structure to expand the library for varying types of APIs and do a cleaner split between concepts.
  

this is just food for thought, the goal here is to improve the code for readability and dev experience in a simple, pragmatic way.

Reply:
I definitely agree some things like this could be approached differently. In all of the ways suggested above. These things have been on my mind for quite some time. I believe the reason we went for `query_id` is that some of these low level methods are never actually used by the end user and none of the additional query attributes are used. Most consumers of the client are usually calling refresh.

Maybe it can be made compatible with multiple approaches (i.e. introduce some method overrides where users can pass Query object or `query_id`). I would have to think a bit further if we can extend the functionality to the Query object itself in a backwards compatible way, but I would be wiling to give it a try.